fe/fe-core/src/main/java/org/apache/doris/datasource/property/fileformat/FileFormatProperties.java

```diff
@@ -0,0 +1,116 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.property.fileformat;
+
+import org.apache.doris.datasource.property.constants.CsvProperties;
+import org.apache.doris.datasource.property.constants.FileFormatBaseProperties;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.proto.InternalService.PFetchTableSchemaRequest;
+import org.apache.doris.thrift.TFileAttributes;
+import org.apache.doris.thrift.TFileCompressType;
+import org.apache.doris.thrift.TFileFormatType;
+import org.apache.doris.thrift.TResultFileSinkOptions;
+import org.apache.doris.thrift.TTextSerdeType;
+
+import java.util.Map;
+
+public abstract class FileFormatProperties {
+    protected TFileFormatType fileFormatType;
+
+    protected TFileCompressType compressionType;
+
+    public FileFormatProperties(TFileFormatType fileFormatType) {
+        this.fileFormatType = fileFormatType;
+    }
+
+    /**
+     *
+     * @param formatProperties
+     * @return properties needed by Doris
+     * @throws AnalysisException
+     */
+
+    /**
+     * Analyze user properties
+     * @param formatProperties properties specified by user
+     * @param isRemoveOriginProperty if this param is set to true, then this method would remove the origin property
+     * @throws AnalysisException
+     */
+    public abstract void analyzeFileFormatProperties(
+            Map<String, String> formatProperties, boolean isRemoveOriginProperty)
+            throws AnalysisException;
+
+    public abstract PFetchTableSchemaRequest toPFetchTableSchemaRequest();
+
+    public abstract TResultFileSinkOptions toTResultFileSinkOptions();
+
+    public abstract TFileAttributes toTFileAttributes();
+
+    public static FileFormatProperties createFileFormatProperties(String formatString) {
```

Comment:
Sometimes we may use the suffix of a file to guess its format.
For example, the file path is `/path/to/1.parquet`, so we know that this is a parquet format, even if user does not specify the "file_format" property.
We should handle this.
And I think without the suffix and user specified "file_format" property, we should provide a "default" format file, eg, csv. But it depends because I am not sure what the previous logic is

Reply:
We can adjust this method when we encount these situations in the next few PRs.